packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -134,25 +134,10 @@ export class ScopeHoistingPackager {
       this.bundle.env.isLibrary ||
       this.bundle.env.outputFormat === 'commonjs'
     ) {
-      for (let b of this.bundleGraph.getReferencedBundles(this.bundle)) {
-        let entry = b.getMainEntry();
-        let symbols = new Map();
-        if (entry && !this.isAsyncBundle && entry.type === 'js') {
-          this.externalAssets.add(entry);
-
-          let usedSymbols = this.bundleGraph.getUsedSymbols(entry) || new Set();
-          for (let s of usedSymbols) {
-            // If the referenced bundle is ESM, and we are importing '*', use 'default' instead.
-            // This matches the logic below in buildExportedSymbols.
-            let imported = s;
-            if (imported === '*' && b.env.outputFormat === 'esmodule') {
-              imported = 'default';
-            }
-            symbols.set(imported, this.getSymbolResolution(entry, entry, s));
-          }
-        }
-
-        this.externals.set(relativeBundlePath(this.bundle, b), symbols);
+      for (let b of this.bundleGraph.getReferencedBundles(this.bundle, {
+        recursive: false,
+      })) {
+        this.externals.set(relativeBundlePath(this.bundle, b), new Map());
       }
     }
```

Comment:
This logic ensures that we add imports for all referenced bundles. But we now wait to determine which symbols to import until they are used (below in `addExternal`). This way, we only include symbols user _in this bundle_ instead of across all bundles.

This also switches to only including bundles non-recursively, otherwise we'd end up hoisting imports for things like CSS even if the library had `"sideEffects": false`. For a library, there's no need to hoist even with side effects. We leave that to the application bundler consuming the library.